HHS/Head-Start-TTADP
# Add file upload api

**Description of change**
This PR adds the a API endpoint POST /files. This is a WIP. It currently has the api endpoint and s3 upload functionality. I would like a checkin to make sure that this approach is sound and makes sense within the context of this project.

***Still to do***
2. Add tests around model
3. Document API endpoint
4. Wire up existing React Components on frontend

**How to test**
1. Update .env file with new variables: 
    1. `AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY`
    2. `AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE`
    3. `MINIO_ROOT_PASSWORD=$AWS_SECRET_ACCESS_KEY`
    4. `MINIO_ROOT_USER=$AWS_ACCESS_KEY_ID`
    5. `S3_BUCKET=ttadp-test`
2. Start app using `yard docker:start`
2. Go to localhost:9000 to access the Minio web console (use AWS_SECRET_ACCESS_KEY for password and AWS_ACCESS_KEY_ID for username).
3. create a bucket named `ttadp-test` (I would like to automate this step in the future)
4. Open a terminal and type `curl localhost:3000/api/files -F File=@TESTFILE.txt -F reportID=123456` where TESTFILE.txt is a file in the directory you run the curl command from
5. Output should be similar to  
```
{
"id":5,"originalFileName":"TESTFILE.txt",
"key":"998cf9ae-7a58-4af5-9bd4-848b08a3b70.txt",
"updatedAt":"2021-01-18T18:41:41.825Z",
"createdAt":"2021-01-18T18:41:41.825Z"
}
```

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/150
